src/main/java/edu/kis/vh/nursery/FIFORhymer.java

```diff
@@ -2,19 +2,24 @@
 
 public class FIFORhymer extends DefaultCountingOutRhymer {
 
-    public DefaultCountingOutRhymer temp = new DefaultCountingOutRhymer();
+    private DefaultCountingOutRhymer temp = new DefaultCountingOutRhymer();
 
     @Override public int countOut() {
         while (!callCheck())
 
-            temp.countIn(super.countOut());
+            getTemp().countIn(super.countOut());
 
-        int ret = temp.countOut();
+        int ret = getTemp().countOut();
 
-        while (!temp.callCheck())
+        while (!getTemp().callCheck())
 
-            countIn(temp.countOut());
+            countIn(getTemp().countOut());
 
         return ret;
     }
+
+    public DefaultCountingOutRhymer getTemp() {

```

Comment:
Shouldn't be encapsulated (typical private field). This is an exaggeration. Using private fields internally usually do not require encapsulation.